random.c

```diff
@@ -829,7 +839,7 @@ static VALUE
 rb_f_srand(int argc, VALUE *argv, VALUE obj)
 {
     VALUE seed, old;
-    rb_random_mt_t *r = &default_rand;
+    rb_random_mt_t *r = rand_mt_start(default_rand());
```

Comment:
`default_mt()`?

Reply:
If we call `rand_mt_start` here, will `rand_init` be called twice when `argc > 0`?

Reply:
Without it, the previous seed will be `false` (`Qfalse`) and srand() returns `false` if there is no initialization.
If returning `false` by `srand()`, we can delete it.

Reply:
Since `rand_init()` should initialize `r->mt`, `genrand_initialized` should be false and it will not be called again.

Reply:
`default_rand` allocates the memory of the default random object at the first `srand` call in each ractor.
Then `random_mt_start` calls `rand_init` for the default random object.
The second `rand_init` is called [below](https://github.com/ruby/ruby/pull/3813/files#diff-32cd45cbdc2d400fa3c2d0f7889e737f7837404a0bf31692c9e7ef014cfba568R851).

Reply:
I see.
Seems OK to return `0` when `old == Qfalse`, instead of initialization here.

Reply:
OK, I'll fix it.